subo/util/token.go

```diff
@@ -2,36 +2,44 @@ package util
 
 import (
 	"io/ioutil"
-	"os"
 	"path/filepath"
 
 	"github.com/pkg/errors"
 )
 
-func getTokenTmpDir() string {
-	tokenPath := filepath.Join(os.TempDir(), "suborbital", "compute", "envtoken")
-	return tokenPath
+func getTokenPath() (string, error) {
+	tokenPath, err := CacheDir("compute")
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(tokenPath, "envtoken"), nil
 }
 
 func WriteEnvironmentToken(tokenStr string) error {
-	tokenPath := getTokenTmpDir()
-	if _, err := os.Stat(tokenPath); os.IsNotExist(err) {
-		if err := os.MkdirAll(filepath.Dir(tokenPath), PermDirectoryPrivate); err != nil {
-			return errors.Wrap(err, "failed to Mkdir")
-		}
+	tokenPath, err := getTokenPath()
+	if err != nil {
+		return err
 	}
 
 	if err := ioutil.WriteFile(tokenPath, []byte(tokenStr), PermFilePrivate); err != nil {
-		return errors.Wrap(err, "failed to WriteFile for token")
+		return errors.Wrap(err, "failed to write "+tokenPath)
```

Comment:
```suggestion
		return errors.Wrapf(err, "failed to write %s " + tokenPath)
```

Reply:
1. I see this as a feature, not a bug! :^) The previous /var/tmp directory location (on macOS anyway) was also per-user. 

2. I don't think handling multiple environments from `subo`'s perspective is a major concern at this point, since the actual token is copied locally into its own file per `compute deploy` folder (and can be overridden). I agree it's not ideal though.

> The current code seems to overwrite any existing token in the cache (though I may have missed where we check for that).

Yes. If you manually create new a token it will overwrite the old one. The other code path would only be triggered if one wasn't cached.

Reply:
Ryan is correct we'll need to handle multiple environments eventually, but it's probably a few months out, and I don't think it'll be a terribly hard change to make. That said, if you want to incorporate that now, I wouldn't be mad.